go/vt/vtgate/planbuilder/operators/insert_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operators
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/plancontext"
+	"vitess.io/vitess/go/vt/vtgate/semantics"
+)
+
+func TestModifyForAutoinc(t *testing.T) {
+	var testCases []string = []string{
+		"insert into test (customer_id, order_date, total_amount, string) values (11, '2023-09-02', 200.00);",
+		"insert into test (customer_id, order_date, total_amount) values (11, '2023-09-02', 200.00, 'test');",
+	}
+	for _, tc := range testCases {
+		t.Run(tc, func(t *testing.T) {
+			ast, err := sqlparser.Parse(tc)
+			require.NoError(t, err)
+
+			insert := ast.(*sqlparser.Insert)
+			_, err = semantics.Analyze(insert, "", &semantics.FakeSI{})
+			require.NoError(t, err)
+
+			ctx := &plancontext.PlanningContext{SemTable: semantics.EmptySemTable()}
+
```

Comment:
This is not correct. You can't throw away the semtable you get from the Analyze phase. 
It has to be kept in the planning context, because without it, the following code can't find the table in the semTable.

Even this won't be enough to fix the test, because you're using an empty FakeSI as the Vschema, which means that the Vschema doesn't actually have the vindex table for the `test` table in it.

It seems to me that instead of adding a test here, it would be easier to just add a test in `plan_test.go` and the corresponding test cases in `dml_cases.json`

Reply:
Okay. Do I have to create a new table for this ?
The queries in `dml_cases.json` are already using 2-3 tables.

Reply:
I added the tests in `dml_cases.json`. However, I could not find the SQL definition of unsharded table. And the test is failing for some reason.